packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/entity_type/mod.rs

```diff
@@ -52,6 +157,78 @@ impl<C: AsClient> EntityTypeStore for PostgresStore<C> {
         Ok(identifier)
     }
 
+    async fn get_entity_type(
+        &self,
+        query: &Expression,
+        data_type_resolve_depth: u8,
+        property_type_resolve_depth: u8,
+        link_type_query_depth: u8,
+        entity_type_query_depth: u8,
+    ) -> Result<Vec<EntityTypeTree>, QueryError> {
+        stream::iter(Read::<PersistedEntityType>::read(self, query).await?)
+            .then(|entity_type: PersistedEntityType| async {
+                let mut data_type_references = HashMap::new();
```

Comment:
Just want to flag a concern with this approach, although I don't think we should change it because I believe the better solution is a caching layer between this and the DB. (Or even some proper in-memory cache we can implement ourselves).

I believe with this approach we could potentially be querying the same data/property types over and over if the stream of entity types refer to the same ones. These hashmaps are basically local caching, but sort of hackily implemented, and only per-type. They do at least better align with the exact structure of the return values, rather than needing to maintain a HashMap per top-level query and then a set per element of that query which we then need to extract _from_ the HashMap...

Reply:
@Alfred-Mountfield @TimDiekmann Is this captured in ticket-form anywhere?

Reply:
Just had a dig, it would seem we do not have a task for a caching layer in general. It's definitely on the cards at some point, but I think designing it or starting work on it now would be premature optimisation and a mistake (Not saying you're suggesting to do so).

Could make a general `Discovery: Cache Layer` or something task to refer to if we need to, but I doubt we'll have much to write in it at this time. (And wouldn't put super specific thoughts like this one in the task I think. We'd revisit this code if profiling suggested it was slow)